great_expectations/core/expectation_suite.py

```diff
@@ -669,29 +672,204 @@ def add_expectation(
+        )
+
     def get_table_expectations(self) -> List[ExpectationConfiguration]:
         """Return a list of table expectations."""
-        return [
-            e
-            for e in self.expectations
-            if e.expectation_type.startswith("expect_table_")
-        ]
+        expectation_configurations: List[ExpectationConfiguration] = list(
+            filter(
+                lambda element: element.get_domain_type() == MetricDomainTypes.TABLE,
+                self.expectations,
+            )
+        )
+
+        expectation_configuration: ExpectationConfiguration
+        for expectation_configuration in expectation_configurations:
+            expectation_configuration.kwargs = deep_filter_properties_iterable(
+                properties=expectation_configuration.kwargs, clean_falsy=True
+            )
+
+        return expectation_configurations
 
     def get_column_expectations(self) -> List[ExpectationConfiguration]:
         """Return a list of column map expectations."""
-        return [e for e in self.expectations if "column" in e.kwargs]
+        expectation_configurations: List[ExpectationConfiguration] = list(
+            filter(
+                lambda element: element.get_domain_type() == MetricDomainTypes.COLUMN,
+                self.expectations,
+            )
+        )
 
+        expectation_configuration: ExpectationConfiguration
+        kwargs: dict
+        column_name: str
+        for expectation_configuration in expectation_configurations:
+            kwargs = deep_filter_properties_iterable(
+                properties=expectation_configuration.kwargs, clean_falsy=True
+            )
+            column_name = kwargs.pop("column")
+            expectation_configuration.kwargs = {"column": column_name, **kwargs}
+
+        return expectation_configurations
+
+    # noinspection PyPep8Naming
     def get_column_pair_expectations(self) -> List[ExpectationConfiguration]:
         """Return a list of column_pair map expectations."""
-        return [
-            e
-            for e in self.expectations
-            if "column_A" in e.kwargs and "column_B" in e.kwargs
-        ]
+        expectation_configurations: List[ExpectationConfiguration] = list(
+            filter(
+                lambda element: element.get_domain_type()
+                == MetricDomainTypes.COLUMN_PAIR,
+                self.expectations,
+            )
+        )
+
+        expectation_configuration: ExpectationConfiguration
+        kwargs: dict
+        column_A_name: str
+        column_B_name: str
+        for expectation_configuration in expectation_configurations:
+            kwargs = deep_filter_properties_iterable(
+                properties=expectation_configuration.kwargs, clean_falsy=True
+            )
+            column_A_name = kwargs.pop("column_A")
+            column_B_name = kwargs.pop("column_B")
+            expectation_configuration.kwargs = {
```

Comment:
Similar to my above comment, could we do `expectation_configuration.kwargs = kwargs` and avoid popping off `column_[AB]`?

Reply:
This implements making it first.